src/python/pants/backend/python/util_rules/faas.py

```diff
@@ -430,16 +430,29 @@ async def infer_runtime_platforms(request: RuntimePlatformsRequest) -> RuntimePl
         file_name = next(
             rt.file_name()
             for rt in request.runtime.known_runtimes
-            if version == (rt.major, rt.minor)
-            and (request.architecture is None or request.architecture.value == rt.architecture)
+            if version == (rt.major, rt.minor) and request.architecture.value == rt.architecture
         )
     except StopIteration:
-        # Not a known runtime, so fallback to just passing a platform
-        # TODO: maybe this should be an error, and we require users to specify
-        # complete_platforms themselves?
-        return RuntimePlatforms(
-            interpreter_version=version,
-            pex_platforms=PexPlatforms((_format_platform_from_major_minor(*version),)),
+        # Not a known runtime, so error out.
+        raise ValueError(
```

Comment:
For the follow-up, mimicking `_infer_from_ics`:

```suggestion
        raise InvaidTargetException(
```